### PR TITLE
feat: add --ignore cmdline option in publish

### PR DESCRIPTION
**Pull request by douglascaetano**
This command line option allows one to specify globs to ignore publishing of matching files. The glob pattern is the same as of gitignore.

### CODE REVIEW

**Comment by douglascaetano** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/cmd/edge_applications/publish/publish.go [hunk 1]
```diff
@@ -89,6 +90,7 @@ func NewCobraCmd(publish *PublishCmd) *cobra.Command {
 	}
 	publishCmd.Flags().BoolP("help", "h", false, msg.EdgeApplicationsPublishFlagHelp)
 	publishCmd.Flags().StringVar(&Path, "path", "public", msg.EdgeApplicationPublishPathFlag)
+	publishCmd.Flags().StringSliceVarP(&Ignore, "ignore", "i", []string{}, msg.EdgeApplicationPublishIgnoreFlag)
```
Comment (PatrickMenoti) [inline pkg/cmd/edge_applications/publish/publish.go:93]: as a convention, we are only using shorthand for identifying flags (such as ID, UUID, etc...)

File: pkg/cmd/edge_applications/publish/get_file_list.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"strings"
+
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+// Return a list of files inside the given directory, ignoring the given globs
+func (cmd *PublishCmd) getFileList(root string, ignore_globs []string) ([]string, error) {
+	var files []string
+
+	// parse ignore globs with gitignore.Pattern.ParsePattern
```
Comment (PatrickMenoti) [inline pkg/cmd/edge_applications/publish/get_file_list.go:15]: do you believe an explanation to the user is necessary regarding this pattern? or we just assume they'll know?

Comment (douglascaetano) [reply pkg/cmd/edge_applications/publish/get_file_list.go:15]: good point! I'll add something

File: pkg/cmd/edge_applications/publish/publish.go (hunk 1 above)
Comment (douglascaetano) [reply pkg/cmd/edge_applications/publish/publish.go]: Done! ✅

File: pkg/cmd/edge_applications/publish/get_file_list.go (hunk 2 above)
Comment (PatrickMenoti) [reply pkg/cmd/edge_applications/publish/get_file_list.go:15]: Perfect!! I'm doing some local tests with this branch now :)

**Review by PatrickMenoti — APPROVED** [review]

**Review by maxwelbm — APPROVED** [review]

**Pull request closed without merging**